static/js/components/SpatialCatalogPage.jsx

```diff
@@ -0,0 +1,430 @@
+    pageNumber,
+    numPerPage,
+    sortData,
+    filterData
+  ) => {
+    setSourcesRowsPerPage(numPerPage);
+    const data = {
+      ...filterData,
+      pageNumber,
+      numPerPage,
+    };
+    if (sortData && Object.keys(sortData).length > 0) {
+      data.sortBy = sortData.name;
+      data.sortOrder = sortData.direction;
+    }
+    dispatch(
+      sourcesActions.fetchSpatialCatalogSources(
+        spatialCatalogName,
+        spatialCatalogEntryName,
+        data
+      )
+    );
+  };
+
+  // eslint-disable-next-line
+  if (!spatialCatalogSources || spatialCatalogSources?.sources?.length === 0) {
+    return (
+      <div className={classes.noSources}>
+        <Typography variant="h5">Entry sources</Typography>
+        <br />
+        <Typography variant="h5" align="center">
+          No sources within entry localization.
+        </Typography>
+      </div>
+    );
+  }
+
+  return (
+    <div className={classes.sourceList}>
+      <SourceTable
+        sources={spatialCatalogSources.sources}
+        title="Spatial Catalog Sources"
+        paginateCallback={handleSourcesTablePagination}
+        pageNumber={spatialCatalogSources.pageNumber}
+        totalMatches={spatialCatalogSources.totalMatches}
+        numPerPage={spatialCatalogSources.numPerPage}
+        sortingCallback={handleSourcesTableSorting}
+        favoritesRemoveButton
+        hideTitle
+      />
+    </div>
+  );
+};
+
+SpatialCatalogSourcesPage.propTypes = {
+  selectedSpatialCatalogId: PropTypes.number.isRequired,
+  selectedSpatialCatalogEntryId: PropTypes.number.isRequired,
+  spatialCatalogs: PropTypes.arrayOf(
+    PropTypes.shape({
+      catalog_name: PropTypes.string,
+      catalog_count: PropTypes.number,
+    })
+  ).isRequired,
+  spatialCatalog: PropTypes.shape({
+    entries: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number,
+        entry_name: PropTypes.string,
+        data: PropTypes.objectOf(PropTypes.any).isRequired, // eslint-disable-line react/forbid-prop-types,
+      })
+    ),
+  }).isRequired,
+};
+
+const SpatialCatalogList = ({ catalogs, deletePermission }) => {
+  const dispatch = useDispatch();
+  const classes = useStyles();
+  const textClasses = textStyles();
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [catalogToDelete, setCatalogToDelete] = useState(null);
+  const openDialog = (id) => {
+    setDialogOpen(true);
+    setCatalogToDelete(id);
+  };
+  const closeDialog = () => {
+    setDialogOpen(false);
+    setCatalogToDelete(null);
+  };
+
+  const deleteCatalog = () => {
+    dispatch(spatialCatalogsActions.deleteSpatialCatalog(catalogToDelete)).then(
+      (result) => {
+        if (result.status === "success") {
+          dispatch(
+            showNotification("Spatial catalog deleting... please be patient.")
+          );
+          dispatch(spatialCatalogsActions.fetchSpatialCatalogs());
+          closeDialog();
+        }
+      }
+    );
+  };
```

Comment:
Very minor comment: on delete, the entries of the deleted catalog still show up until your refresh or select another one. That's ok though, and can be fixed in a following PR.

Reply:
yeah it's because it's async.